cannon/multicannon/run.go

```diff
@@ -0,0 +1,21 @@
+package main
+
+import (
+	"os"
+
+	"github.com/urfave/cli/v2"
+
+	"github.com/ethereum-optimism/optimism/cannon/cmd"
+	"github.com/ethereum-optimism/optimism/cannon/mipsevm/versions"
+)
+
+func Run(ctx *cli.Context) error {
+	inputPath := ctx.Path(cmd.RunInputFlag.Name)
+	version, err := versions.DetectVersion(inputPath)
+	if err != nil {
+		return err
+	}
+	return ExecuteCannon(os.Args[1:], version)
+}
+
+var RunCommand = cmd.CreateRunCommand(Run)
```

Comment:
I wonder if we can make this more generic so that it only parses the key args it needs and passes through the other unchanged.  Otherwise we're going to have compatibility issues if we ever change the CLI args in the future because the older cannon versions will have different args to the new ones.  If multicannon only actually parses the minimum number of flags it needs to do its job then we can change any other args and it will keep working.  Whereas if we always reuse the options from the latest version we're likely to get stuck.

Reply:
The CLI possibly breaking is a salient point. Lemme try something.

Reply:
Tweaked this to forward flags without inspecting them, except `--input` and `--type` depending on the command.
One downside is that `--help` breaks unless a valid `--input` or `--type` is specified.
And unconditionally forwarding the `--help` isn't an option as we need to first select the VM implementation (using input/type) for the help display. No big deal though. Users can always use the VM implementation CLI directly if they need `--help`.

Reply:
That is a bit unfortunate. Could we have `--help` by itself print the list of supported versions and a note to say use `--type x --help`

Reply:
Good idea.